### PR TITLE
[TEVA-2057] Send job application received emails to publishers

### PR DESCRIPTION
**Pull request by cesidio**
## Jira ticket URL
https://dfedigital.atlassian.net/browse/TEVA-2057

## Screenshots of UI changes:
Screenshot 2021-04-07 at 20 05 21

### CODE REVIEW

**Comment by martin-bangoura** [comment]
Review app deployed to <https://teaching-vacancies-review-pr-3224.london.cloudapps.digital>

File: app/controllers/publishers/base_controller.rb [hunk 1]
```diff
@@ -3,6 +3,6 @@ class Publishers::BaseController < ApplicationController
                 :check_terms_and_conditions
 
   def check_terms_and_conditions
-    redirect_to terms_and_conditions_path unless current_publisher.accepted_terms_and_conditions?
+    redirect_to terms_and_conditions_path unless current_publisher.accepted_terms_at?
```
Comment (cpjmcquillan) [inline app/controllers/publishers/base_controller.rb:6]: Can I be extra cheeky and ask to put this mini 🧹 🔥  in a separate PR?

File: app/mailers/publishers/job_application_mailer.rb [hunk 2]
```diff
@@ -0,0 +1,15 @@
+class Publishers::JobApplicationMailer < Publishers::BaseMailer
+  helper VacanciesHelper
+
+  def applications_received(publisher:)
+    @template = NOTIFY_PUBLISHER_JOB_APPLICATIONS_RECEIVED_TEMPLATE
+    @publisher = publisher
+    @vacancies = publisher.vacancies_with_job_applications_submitted_yesterday
+    @to = publisher.email
+
+    @job_applications_count = @vacancies.sum { |vacancy| vacancy.job_applications.count }
```
Comment (cpjmcquillan) [inline app/mailers/publishers/job_application_mailer.rb:10]: Perhaps we want the vacancy id and the job applications data to be sent in the `email_event_data`?

Is it worth adding something for now, and then checking with @stevenleggdfe what he might need when he's back from leave?

File: app/controllers/publishers/base_controller.rb (hunk 1 above)
Comment (cesidio) [reply app/controllers/publishers/base_controller.rb:6]: of course you can!

File: app/models/job_application.rb [hunk 3]
```diff
@@ -34,6 +34,8 @@ class JobApplication < ApplicationRecord
   has_many :employments, dependent: :destroy
   has_many :references, dependent: :destroy
 
+  scope :submitted_yesterday, -> { submitted.where("DATE(submitted_at) = ?", Date.yesterday) }
```
Comment (cpjmcquillan) [inline app/models/job_application.rb:37]: Do we use this scope anywhere?

File: spec/jobs/send_applications_received_yesterday_job_spec.rb [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+
+RSpec.describe SendApplicationsReceivedYesterdayJob do
+  subject(:job) { described_class.perform_later }
+
+  let(:publisher) { create(:publisher) }
+  let(:message_delivery) { instance_double(ActionMailer::MessageDelivery) }
+
+  before { expect(Publisher).to receive_message_chain(:distinct, :joins, :where).and_return([publisher]) }
+
+  it "sends applications received emails" do
```
Comment (cpjmcquillan) [inline spec/jobs/send_applications_received_yesterday_job_spec.rb:11]: Maybe add a spec for the triggered email event with data (if you add it)

**Review by cpjmcquillan — APPROVED** [review]
Couple of comments but looks good to me!

File: app/models/job_application.rb (hunk 3 above)
Comment (cesidio) [reply app/models/job_application.rb:37]: yes we use it in the email:
```
  <%= notify_link(organisation_job_job_applications_url(vacancy),
                  t(".view_applications", count: vacancy.job_applications.submitted_yesterday.count)) %>
```

**Review by cpjmcquillan — APPROVED** [review]

**Pull request merged**